index.js

```diff
@@ -11,26 +11,58 @@ const nextButton = document.querySelector('[data-js="button-next"]');
 const pagination = document.querySelector('[data-js="pagination"]');
 
 // States
-const maxPage = 1;
-const page = 1;
+let maxPage = 1; // changed to let
+let page = 1;
 const searchQuery = "";
 
-async function fetchCharacters() {
-  const response = await fetch("https://rickandmortyapi.com/api/character");
-  const data = await response.json();
-
-  return data.results;
+async function fetchCharacters(page) {
```

Comment:
You don't need the parameter here. Your function can access the global variable 'page'.